parser_utils.py

```diff
@@ -0,0 +1,331 @@
+from enum import Enum
+
+class Lists(Enum):
+    """Enum class for representing various list types that appear in the grammar."""
+    GOPLIST = 0
+    ANYLIST = 1
+    IDLIST = 2
+    MIXEDLIST = 3
+    EXPLIST = 4
+
+
+class Ops(Enum):
+    """Enum class for representing various Op types that appear in the grammar."""
+    GATE = 0
+    OPERATOR = 1
+    CHANNEL = 2
+    MEASURE = 3
+    BARRIER = 4
+
+
+class Declarations(Enum):
+    """Enum class for representing various declarations that are allowed by the grammar."""
+    QREG = 0
+    CREG = 1
+    GATE = 2
+    OPERATOR = 3
+    CHANNEL = 4
```

Comment:
I figured this kind of structure would make things more easily extendible than empty subclasses